model_analyzer/entrypoint.py

```diff
@@ -296,24 +292,49 @@ def main():
     Main entrypoint of model_analyzer
     """
 
+    global exiting
+
+    # Model Analyzer Status
+    exiting = False
+
     monitoring_metrics = [
         PerfThroughput, GPUUtilization, GPUUsedMemory, GPUFreeMemory
     ]
 
+    def interrupt_handler(signal, frame):
+        global exiting
+        exiting = True
+        logging.info('Received SIGINT. Exiting...')
+        return
+
+    signal.signal(signal.SIGINT, interrupt_handler)
+
     try:
         args = CLI().parse()
     except TritonModelAnalyzerException as e:
-        print(f'Model Analyzer encountered an error : {e}')
+        logging.error(f'Model Analyzer encountered an error: {e}')
 
+    logging.info(f'Triton Model Analyzer started {args} arguments')
+    analyzer = Analyzer(args, monitoring_metrics)
     try:
-        analyzer = Analyzer(args, monitoring_metrics)
+        logging.info('Starting Triton Server...')
         client, server = get_triton_handles(args)
+        logging.info('Triton Server launched.')
+
+        # Only check for exit after the events that take a long time.
+        if exiting:
+            return
+
         check_triton_and_model_analyzer_gpus(args)
         run_configs = create_run_configs(args)
+        if exiting:
```

Comment:
The keyboard interrupt can be caught and propagated up until this level, where it can be caught.